src/main/kotlin/uk/gov/gdx/datashare/config/ResourceServerConfiguration.kt

```diff
@@ -30,26 +31,48 @@ class ResourceServerConfiguration {
       .roles(role)
   }
 
+  @Order(1)
   @Bean
-  fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
+  fun permittedFilterChain(http: HttpSecurity): SecurityFilterChain {
     return http
       .csrf { it.disable() } // csrf not needed on a rest api
-      .authorizeHttpRequests { it ->
-        it.requestMatchers(
-          "/favicon.ico",
-          "/health/**",
-          "/info",
-          "/v3/api-docs/**",
-          "/v3/api-docs.yaml",
-          "/swagger-ui/**",
-          "/swagger-ui.html",
-          "/metrics/**",
-        )
-          .permitAll()
-        it.requestMatchers("/prometheus").hasRole("ACTUATOR").and().httpBasic()
+      .securityMatcher(
+        "/favicon.ico",
+        "/health/**",
+        "/info",
+        "/v3/api-docs/**",
+        "/v3/api-docs.yaml",
+        "/swagger-ui/**",
+        "/swagger-ui.html",
+        "/metrics/**",
+      )
+      .authorizeHttpRequests {
+        it.anyRequest().permitAll()
+      }
+      .build()
+  }
+
+  @Order(2)
+  @Bean
+  fun prometheusFilterChain(http: HttpSecurity): SecurityFilterChain {
+    return http
+      .csrf { it.disable() }
+      .securityMatcher("/prometheus")
+      .authorizeHttpRequests {
+        it.anyRequest().hasRole("ACTUATOR")
+      }
+      .httpBasic {}
+      .build()
+  }
+
+  @Bean
+  fun apiFilterChain(http: HttpSecurity): SecurityFilterChain {
```

Comment:
Does no Order on this matter?

Reply:
Was just copied from the docs - happy to add LAST

Reply:
It's `Ordered.LOWEST_PRECEDENCE` which is `Integer.MAX_VALUE` under the hood 😄